File: Bit Manipulation/Missing_Number/Missing_Number.java
class Missing_Number {
    public int missingNumber(int[] nums) {
        int n = nums.length;
        int xor = 0;
        for (int i = 0; i <= n; i++) {
            System.out.println("xor value before ->"+xor);
            xor ^= i;
            System.out.println("i value -> "+i +" xor value after ->"+xor);
        }

        for (int num : nums) {
            xor ^= num;
            System.out.println("num  value ->  "+num +" xor value -> "+xor);
        }

        return xor;
    }
    public static void main(String[] args) {
        Missing_Number obj = new Missing_Number();
        int[] nums = { 0, 1};
        System.out.println(obj.missingNumber(nums));
    }
}

File: Bit Manipulation/Missing_Number/Missing_Number.py
def missingNumber(nums):
    n=len(nums)
    total=n*(n+1)//2
    print(total)
    sum_of_nums=sum(nums)
    print(sum_of_nums)
    return total-sum_of_nums

def main():
    nums=[3,0,1]
    print(missingNumber(nums))
if __name__ == "__main__":
    main()